api/client/client.go

```diff
@@ -266,3 +276,21 @@ func NewAuthServer(addr string, provider auth.Provider) *AuthRemote {
 func (ar *AuthRemote) Sign(req []byte) ([]byte, error) {
 	return ar.AuthSign(req, nil, ar.provider)
 }
+
+// nomalizeURL checks for http/https protocol, appends "http" as default protocol if not defiend in url
+func normalizeURL(addr string) string {
+	addr = strings.TrimSpace(addr)
+	u, err := url.Parse(addr)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if !strings.Contains(u.Scheme, "http") {
```

Comment:
Actually, the problem with that example is that URI Scheme's must begin with a letter (so an IP address will never match a Scheme). However, a DNS name can `"ca.example.com:8888"` will parse incorrectly

You can instead check if `u.Opaque` has been set (it never should be for a normal URL) https://play.golang.org/p/3-8Yeu4Ily